File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        i = 0
        ans = 0

        romans = {
            "I" : 1,
            "V" : 5,
            "X" : 10,
            "L" : 50,
            "C" : 100,
            "D" : 500,
            "M" : 1000
        }

        while i < len(s):
            if i == len(s) - 1:
                ans += romans[s[i]]
                i += 1
            
            elif romans[s[i]] < romans[s[i + 1]]:
                ans += romans[s[i + 1]] - romans[s[i]]
                i += 2
            
            else:
                ans += romans[s[i]]
                i += 1
        
        return ans

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        int i = 0, ans = 0;
        Map<Character, Integer> romans = new HashMap<>();

        romans.put('I', 1);
        romans.put('V', 5);
        romans.put('X', 10);
        romans.put('L', 50);
        romans.put('C', 100);
        romans.put('D', 500);
        romans.put('M', 1000);

        while (i < s.length()){
            if(i == s.length() - 1){
                ans += romans.get(s.charAt(i));
                i++;
            }
            else if(romans.get(s.charAt(i)) < romans.get(s.charAt(i+1))){
                ans += romans.get(s.charAt(i + 1)) - romans.get(s.charAt(i));
                i+=2;
            }
            else{
                ans += romans.get(s.charAt(i));
                i++;
            }
            
        }

        return ans;

    }
}